test/functional/rpc_getblockstats.py

```diff
@@ -44,24 +32,42 @@ def add_options(self, parser):
                             help='Test data file')
 
     def set_test_params(self):
-        self.num_nodes = 2
-        self.extra_args = [['-txindex'], ['-paytxfee=0.003']]
+        self.num_nodes = 1
         self.setup_clean_chain = True
+        self.extra_args = [['-txindex']]
 
     def get_stats(self):
         return [self.nodes[0].getblockstats(hash_or_height=self.start_height + i) for i in range(self.max_stat_pos+1)]
 
     def generate_test_data(self, filename):
-        mocktime = time.time()
+        mocktime = 1525107225
+        self.nodes[0].setmocktime(mocktime)
         self.nodes[0].generate(101)
 
-        self.nodes[0].sendtoaddress(address=self.nodes[1].getnewaddress(), amount=10, subtractfeefromamount=True)
+        pk1 = self.nodes[0].getaddressinfo(self.nodes[0].getnewaddress())['pubkey']
+        pk2 = self.nodes[0].getaddressinfo(self.nodes[0].getnewaddress())['pubkey']
+
+        nested_p2wpkh_address = self.nodes[0].getnewaddress('', 'p2sh-segwit')
+        native_p2wpkh_address = self.nodes[0].getnewaddress('', 'bech32')
+        native_p2wsh_address = self.nodes[0].addmultisigaddress(2, [pk1,pk2], '', 'bech32')['address']
+        nested_p2wsh_address = self.nodes[0].addmultisigaddress(2, [pk1,pk2], '', 'p2sh-segwit')['address']
+
+        # testing batching, nested p2wpkh, dust, native p2wsh, and nested p2wsh metrics
+        outputs = {nested_p2wpkh_address:4,self.nodes[0].getnewaddress('', 'legacy'):0.00001,native_p2wsh_address:2,nested_p2wsh_address:30}
```

Comment:
We create two p2wsh outputs: one native and one nested. In the JSON file only one is counted in `new_p2wsh_outputs`.

https://github.com/bitcoinops/bitcoin/blob/6b2e3eb69ec6206768a89587888b6772ce49175a/test/functional/data/rpc_getblockstats.json#L260

I guess what happens here is that we can't determine if a `p2sh` wraps a `p2w*` script from the `scriptPubKey`. Maybe consider renaming both `new_p2wpkh_outputs` and `new_p2wsh_outputs` to `new_native_p2w(pkh/sh)_outputs` to avoid confusion somewhere down the line.

Reply:
> nit: Add a space or line break after each argument to improve readability.

Done

> Maybe consider renaming...

That's a good suggestion. However, it'd be a breaking API change, and we already have data on our deployed dashboard that uses the current naming.

I'll keep it as it is, but update the help text to make it clear that this is just for the native outputs.